src/target/adi.c

```diff
@@ -565,7 +570,7 @@ static void adi_parse_adi_rom_table(adiv5_access_port_s *const ap, const target_
 	}
 
 	/* Check SYSMEM bit */
-	const bool memtype = adi_mem_read32(ap, base_address + ADIV5_ROM_MEMTYPE) & ADIV5_ROM_MEMTYPE_SYSMEM;
+	const bool memtype = adi_mem_read32(ap, base_address + ADIV5_ROM_CLASS1_MEMTYPE) & ADIV5_ROM_CLASS1_MEMTYPE_SYSMEM;
```

Comment:
As per the note below, we advise using the nomenclature `ADIV5_ROM_ADI_MEMTYPE` here, with `ADIV5_ROM_CORESIGHT_` used for the parser below. "Class 1" ROMs are only defined by the debug interface spec and are not CoreSight ROMs, while "Class 9" are CoreSight ROMs and they have a version number associated with them too (v0 right now, but ARM left space for a v1, v2, etc).